### ISSUE TITLE
Upload from Android app fails

### ISSUE THREAD

**Issue by kvoit** [nextcloud/android#287]
### Actual behaviour

-(Auto)upload from Android app fails ALMOST always after a variable progress
### Expected behaviour

-(Auto)upload from Android app should succeed :)
### Steps to reproduce
1. Configure App to upload new photos (here: in Wifi, without charging)
2. Take photo
3. See upload progress to a variable level, then fail with "connection error" ("Verbindungsfehler")
### Further info

Uploading AND autouploading of certain files fails. The failing seems pretty reproducible for some file, especially most (but not all) camera images. Some other file can be uploaded without problems (including a 400+MB video). Some other videos also fail. I cannot find any pattern. Behavior is observed in 2 generally stable wifis.
I should perhaps note: The file for the logs belowis 4197043 bytes large. The error says it expected 353728 got 101043.
If the last chunk has 101043, this would leave 1024000 bytes for each of the other chunks, with is a number that makes a lot of sense. Why did it expect 353728 bytes?
I am pretty sure this is a very strong pointer to the root of the problem.
### Environment data

Android version: 6 (OxygenOS 3.5.1)

Device model: Oneplus 3

Stock or customized system: Stock (Community release)

Nextcloud app version: 1.3.1-gplay

Nextcloud server version: 10.0
#### Web server error log

access.log:

```
195.37.61.2 - kay [23/Sep/2016:09:56:35 +0200] "HEAD /remote.php/webdav/SofortUpload/2016/09/ HTTP/1.1" 200 1366 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:36 +0200] "HEAD /remote.php/webdav/SofortUpload/2016/09/IMG_20160923_095633.jpg HTTP/1.1" 404 1240 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:36 +0200] "PUT /remote.php/webdav/SofortUpload/2016/09/IMG_20160923_095633.jpg-chunking-19670429-5-0 HTTP/1.1" 201 1261 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:37 +0200] "PUT /remote.php/webdav/SofortUpload/2016/09/IMG_20160923_095633.jpg-chunking-19670429-5-1 HTTP/1.1" 201 1253 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:38 +0200] "PUT /remote.php/webdav/SofortUpload/2016/09/IMG_20160923_095633.jpg-chunking-19670429-5-2 HTTP/1.1" 201 1257 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:39 +0200] "PUT /remote.php/webdav/SofortUpload/2016/09/IMG_20160923_095633.jpg-chunking-19670429-5-3 HTTP/1.1" 201 1253 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:39 +0200] "PUT /remote.php/webdav/SofortUpload/2016/09/IMG_20160923_095633.jpg-chunking-19670429-5-4 HTTP/1.1" 400 1548 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:42 +0200] "GET /index.php/avatar/kay/384 HTTP/1.1" 200 1652 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:42 +0200] "GET /ocs/v1.php/cloud/users/kay?format=json HTTP/1.1" 200 2521 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:42 +0200] "GET /ocs/v1.php/cloud/users/kay?format=json HTTP/1.1" 200 2519 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
195.37.61.2 - kay [23/Sep/2016:09:56:42 +0200] "GET /index.php/avatar/kay/384 HTTP/1.1" 200 1505 "-" "Mozilla/5.0 (Android) ownCloud-android/1.3.1"
```

error.log
_nothing_
#### Nextcloud log (data/nextcloud.log)

```
{"reqId":"rRs5BlHKp0igYvtlOY7J","remoteAddr":"195.37.61.2","app":"webdav","message":"Exception: {\"Message\":\"HTTP\\\/1.1 400 expected filesize 353728 got 101043\",\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(104): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->createFileChunked(Resource id #81)\\n#1 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(137): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #81)\\n#2 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1036): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('IMG_20160923_09...', Resource id #81)\\n#3 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(523): Sabre\\\\DAV\\\\Server->createFile('SofortUpload\\\/20...', Resource id #81, NULL)\\n#4 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#5 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#6 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(459): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#7 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(248): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(60): Sabre\\\\DAV\\\\Server->exec()\\n#9 \\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/remote.php(165): require_once('\\\/var\\\/www\\\/example....')\\n#10 {main}\",\"File\":\"\\\/var\\\/www\\\/example.net\\\/cloud\\\/nextcloud-10.0.0\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":405,\"User\":\"kay\"}","level":4,"time":"2016-09-23T07:56:40+00:00","method":"PUT","url":"\/remote.php\/webdav\/SofortUpload\/2016\/09\/IMG_20160923_095633.jpg-chunking-19670429-5-4","user":"kay"}
```

I know there are a lot of similar bug reports around, with various configuration solutions. I think I have added them all to my configs:

PHP (7.0.8-0ubuntu0.16.04.2, values from phpinfo(), so I know it is active):

```
max_execution_time = 3600
max_input_time = 3600
max_file_uploads = 50
upload_max_filesize = 16G
post_max_size = 16G
upload_tmp_dir = /data/tmp (disk has enough space, tried /tmp on SSD before, too)
memory_limit = 256M
```

Apache (Apache/2.4.18 (Ubuntu)):

```
KeepAlive On
MaxKeepAliveRequests 500
KeepAliveTimeout 500
```

```
<VirtualHost *:443>
    ServerAdmin webmaster@example.net
    ServerName cloud.example.net

    DocumentRoot /var/www/example.net/cloud/current

    <Directory />
        Options +FollowSymLinks -Indexes
        AllowOverride None
    </Directory>
    <Directory /var/www/>
        Options FollowSymLinks MultiViews
        AllowOverride None
        Order allow,deny
        allow from all
    </Directory>
    <Directory /var/www/example.net/cloud>
        SSLRenegBufferSize 10486000

        <Limit GET POST OPTIONS PROPFIND PUT DELETE MKCOL MOVE>
                Order allow,deny
                Allow from all
        </Limit>
        <LimitExcept GET POST OPTIONS PROPFIND PUT DELETE MKCOL MOVE>
                Order deny,allow
                Deny from all
        </LimitExcept>
    </Directory>

    ErrorLog ${APACHE_LOG_DIR}/error.log
    # Possible values include: debug, info, notice, warn, error, crit, alert, emerg.
    LogLevel warn
    CustomLog ${APACHE_LOG_DIR}/ssl_access.log combined

    SSLEngine on
    Header always set Strict-Transport-Security "max-age=15552000; includeSubDomains"

    SSLCertificateFile    /etc/letsencrypt/live/cloud.example.net/fullchain.pem
    SSLCertificateKeyFile /etc/letsencrypt/live/cloud.example.net/privkey.pem

    SSLProtocol All -SSLv2 -SSLv3
    SSLHonorCipherOrder on
    SSLCipherSuite "EECDH+ECDSA+AESGCM EECDH+aRSA+AESGCM EECDH+ECDSA+SHA384 EECDH+ECDSA+SHA256 EECDH+aRSA+SHA384 EECDH+aRSA+SHA256 EECDH EDH+aRSA !RC4 !aNULL !eNULL !LOW !3DES !MD5 !EXP !PSK !SRP !DSS"
</VirtualHost>
```

Nextcloud:

```
<?php
$CONFIG = array (
  'instanceid' => '[...]',
  'passwordsalt' => '[...]',
  'trusted_domains' => 
  array (
    0 => 'cloud.example.net',
    1 => 'owncloud.example.net',
  ),
  'datadirectory' => '/data/programs/nextcloud',
  'dbtype' => 'mysql',
  'version' => '9.1.0.16',
  'dbname' => 'nextcloud',
  'dbhost' => '127.0.0.1',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'nextcloud',
  'dbpassword' => '[...]',
  'installed' => true,
  'theme' => '',
  'maintenance' => false,
  'forcessl' => true,
  'mail_smtpmode' => 'smtp',
  'mail_from_address' => 'owncloud-admin',
  'mail_domain' => 'example.net',
  'mail_smtphost' => 'localhost',
  'mail_smtpport' => '25',
  'overwritewebroot' => '/',
  'secret' => '[...]',
  'loglevel' => 2,
  'overwrite.cli.url' => '/',
  'updatechecker' => false,
  'trashbin_retention_obligation' => 'auto',
  'htaccess.RewriteBase' => '/',
  'memcache.local' => '\\OC\\Memcache\\APCu',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'filelocking.enabled' => 'true',
  'redis' => 
  array (
    'host' => '/var/run/redis/redis.sock',
    'port' => 0,
    'timeout' => 0.0,
  ),
);
```

In the last config I notice the version. Why is this not 10?

```
sudo -u www-data php ./occ upgrade
Nextcloud is already latest version
```

**Comment by kvoit** [comment]
I should perhaps note: The file is 4197043 bytes large. The error says it expected 353728 got 101043.
If the last chunk has 101043, this would leave  1024000 bytes for each of the other chunks, with is a number that makes a lot of sense. Why did it expect 353728 bytes?
I am pretty sure this is a very strong pointer to the root of the problem.

**Comment by kvoit** [comment]
I would really like to try to increase chunk size for further testing, but that does not seem possible.

**Comment by kvoit** [comment]
Please note that this is most probably not a duplicate of #85. There, full chunks with an expected size of 1024000 prematurely fail. Here only last chunks fail, apparently because SabreDAV expects something bigger than (file size - n*1024000).

**Comment by kvoit** [comment]
More info: I tried the same thing on the Sony Z3@Stock of my wife, and I could replicate the problem. A little more in detail, a first run of quite a lot of images works fine. Perhaps 50 to 100 (on both devices, I initially uploaded the DCIM folder with all old images). After those, any further attempt fails with the error above.

**Comment by AndyScherzinger** [comment]
@kvoit the beta is still available and is 5 days old: https://f-droid.org/repository/browse/?fdfilter=com.nextcloud&fdid=com.nextcloud.android.beta

The issue with the last chunk is likely related to the number of chunks and overall file size since the server then puts everything together with the last chunk which sometimes ends with a http timeout on the client side.

**Comment by kvoit** [comment]
Oh, indeed. My F-Droid didn't update properly. I have it installed now. The uploads still fail, the protocol is empty.
I do not think this is a timeout problem. It happens too fast. And it does not happen with large files. A 400MB video works, 4MB pictures regularly fail. I am also not exactly doing this on a supercomputer, but the N36L is still more powerful than a Raspberry Pi, which people seem to use.

Have you taken into account that according to the log, the server receives exactly the size of the image, but for some reason expects more, as stated in the further info section of my report?

**Comment by AndyScherzinger** [comment]
Hi @kvoit sorry, I missed that in the logs. @tobiasKaminsky any idea as to why this happens?

**Comment by ArnaudAUBERT** [comment]
I confirm, i have the same problem on my OnePlus 3.
All was fine, but since a week, upload of photos does not work.
Don't work on server 9.5 so i upgrade to 10.0.
But it don't work on server 10.0.

**Comment by AndyScherzinger** [comment]
@ArnaudAUBERT what is the Operating System version on your OnePlus3 and do you know on which Android version it is based on? Simply asking since at the moment instant uploads are broken for phones running Android Nougat / 7 which is an issue we are currently working on.

**Comment by ArnaudAUBERT** [comment]
It is the last stock version :
Android version : 6.0.1 and OxygenOS version : 3.2.6

**Comment by AndyScherzinger** [comment]
Hmmm, Android 6 should work, at least there haven't been any issues been reported regarding instant upload even though I friend who is still running the ownCloud app reported the same issue with his OnePlus3. Are you by any change running the beta app and have anything related to this matter in the app's log file?

**Comment by ArnaudAUBERT** [comment]
I have Nextcloud App version 1.3.1
Where can I find the app's log file ?

**Comment by AndyScherzinger** [comment]
Unfortunately you can't since the stable releases don't ship logging capabilities, the logging is only available in the f-droid release beta client (which can be installed in parallel)  https://f-droid.org/repository/browse/?fdfilter=com.nextcloud&fdid=com.nextcloud.android.beta

**Comment by ArnaudAUBERT** [comment]
It's a modified version of Android 6.0.1 named OxygenOs version 3.2.6
It was working fine in august but I think it's an OxygenOS update that broke it.

**Comment by AndyScherzinger** [comment]
Okay, in that case we mostly likely need #285 to be implemented to solve this issue

**Comment by ArnaudAUBERT** [comment]
I have installed the last beta version from fdroid but it doest not work too.

**Comment by AndyScherzinger** [comment]
Do you see anything in the app's log? (via settings--> logs (at the bottom of the settings screen)

**Comment by ArnaudAUBERT** [comment]
nothing, it's empty

**Comment by AndyScherzinger** [comment]
hmm, pity. We will have to look into that logging issue too then :(

**Comment by kvoit** [comment]
I can confirm everything ArnaudAUBERT said for my OP3 on the latest Community Edition version 3.5.3. I cannot give you any timeline, since I only switched to Nextcloud and put most of the config in place directly before I submitted the problem.

I DO, though, also have similar problems with the Stock Sony Z3 of my wife.

**Comment by AndyScherzinger** [comment]
@kvoit that' right yes. I also opened a ticket for the logs of the client #297 but couldn't reproduce the issue since it works on the Nexus5X so it seems to be device/os specific :(

**Comment by Happyfeet01** [comment]
another one with the same issue.

_Xperia Z3
Android 6.0.1
unlocked Bootloader._

i don´t know if you this but the **Android log** say this:

`09-29 14:53:31.589 E/AndroidRuntime(413): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nextcloud.client/com.owncloud.android.ui.activity.ReceiveExternalFilesActivity}: java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0`

**here my logfiles from Server:**

`Exception: {"Message":"HTTP\/1.1 404 File with name Sofortupload\/2016\/09\/DSC_0007.JPG could not be located","Exception":"Sabre\\DAV\\Exception\\NotFound","Code":0,"Trace":"#0 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(79): OCA\\DAV\\Connector\\Sabre\\ObjectTree->getNodeForPath('Sofortupload\/20...', 0)\n#1 [internal function]: Sabre\\DAV\\CorePlugin->httpGet(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#2 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#3 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\\Event\\EventEmitter->emit('method:GET', Array)\n#4 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php(251): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response), false)\n#5 [internal function]: Sabre\\DAV\\CorePlugin->httpHead(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#6 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#7 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\\Event\\EventEmitter->emit('method:HEAD', Array)\n#8 \/var\/www\/html\/nextcloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#9 \/var\/www\/html\/nextcloud\/apps\/dav\/appinfo\/v1\/webdav.php(60): Sabre\\DAV\\Server->exec()\n#10 \/var\/www\/html\/nextcloud\/remote.php(165): require_once('\/var\/www\/html\/n...')\n#11 {main}","File":"\/var\/www\/html\/nextcloud\/apps\/dav\/lib\/Connector\/Sabre\/ObjectTree.php","Line":173,"User":"Lars"}`

**Comment by kvoit** [comment]
Though this seems to be a different issue (at least a different log entry, no "expected filesize" problem), I am glad at least now the Z3 as my second problematic device as another person confirming :)
Some updates from my side: Further attempts to upload from failed files fail almost immediately my OP3. The upload list shows "connection error", but there are no error logs in the server, only this access log:
```
195.37.61.2 - kay [29/Sep/2016:16:16:15 +0200] "HEAD /remote.php/webdav/SofortUpload/ HTTP/1.1" 200 1235 "-" "Mozilla/5.0 (Android) nextcloud-android/20160922"
195.37.61.2 - kay [29/Sep/2016:16:16:15 +0200] "HEAD /remote.php/webdav/SofortUpload/IMG_20160709_165052%20(2).jpg HTTP/1.1" 404 1252 "-" "Mozilla/5.0 (Android) nextcloud-android/20160922"
```
Also no client logs.

**Comment by Happyfeet01** [comment]
Again I make try to make an upload..

09-29 20:17:39.616 E/UploadFileOperation(11776): Upload of 
/storage/6237-6530/DCIM/100ANDRO/DSC_0009.JPG to 
/Sofortupload/2016/09/DSC_0009.JPG: Unexpected exception

Gesendet mit AquaMail für Android
http://www.aqua-mail.com


Am 29. September 2016 4:18:26 nachm. schrieb kvoit <notifications@github.com>:

**Comment by gdamjan** [comment]
Same problem for me,
I too have oneplus 3 (with Android 6.0.1 OxygenOS 3.2.6). running nextcloud 10.0.1 with nginx and uwsgi-php. client from f-droid 1.3.1. (I also had the problem with owncloud 9.1.{0,1})

The PUT request never even gets to the application, nginx reports a client timeout while writing the file to its temporary storage.

I'm enabling some more nginx debug logs now (needs recompile with --with-debug)

**Comment by gdamjan** [comment]
it seems the content-length header is wrong
2016/10/03 16:29:48 [debug] 6446#6446: *1 http exten: "jpg-chunking-10646784-3-2"
2016/10/03 16:29:48 [debug] 6446#6446: *1 posix_memalign: 00000000011ECD40:4096 @16
2016/10/03 16:29:48 [debug] 6446#6446: *1 http process request header line
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "OC-Chunked: OC-Chunked"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "OC-Chunk-Size: 157540"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "OC-Total-Length: 2914460"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "Authorization: Basic xxx"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "User-Agent: Mozilla/5.0 (Android) ownCloud-android/1.3.1"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "Host: damjan.ie.mk"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "Content-Length: 1024000"
2016/10/03 16:29:48 [debug] 6446#6446: *1 http header: "Content-Type: image/jpeg"

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky can you have a look at this? Seems @gdamjan has found the cause for this :)

**Comment by iskradelta** [comment]
A possibly fix, if anyone is willing to try it out https://github.com/nextcloud/android-library/pull/20

The content-lenght was wrong for the last chunk when retrying uploads, due to using channel.position to calculate the value - which was not yet moved to the offset it was being read from. So nginx/web-server kept waiting for more data until timeout and connection-close, error logs and so on.

**Comment by AndyScherzinger** [comment]
Awesome finding @iskradelta and also perfect since the PR is targeting beta! :smiley: 
@tobiasKaminsky I think this can be merged to beta to test drive :rocket:

**Comment by kvoit** [comment]
After updating to the 20161019 beta, I seem to have a lot less problems. I still get some failed uploads, but those to not leave any error log on the nextcloud server. They also fail very quickly (<-~1s) with a "connection error" message. Note that this is while successfully uploading 20 other images.

The only trace I could find on the server is this access.log:
```
2001:4dd0:ff00:9bf0:xxx - myuser [24/Oct/2016:07:26:17 +0200] "HEAD /remote.php/webdav/SofortUpload/2016/10/ HTTP/1.1" 200 1229 "-" "Mozilla/5.0 (Android) nextcloud-android/20161019"
2001:4dd0:ff00:9bf0:xxx - myuser [24/Oct/2016:07:26:18 +0200] "HEAD /remote.php/webdav/SofortUpload/2016/10/IMG_20160709_165054.jpg HTTP/1.1" 404 1246 "-" "Mozilla/5.0 (Android) nextcloud-android/20161019"
```

I suspect this is a different problem that the one originally emphasized in this bug report and we need a client protocol to access it.

**Comment by AndyScherzinger** [comment]
Hi @kvoit do you see any errors for this issue in the client log? With the 20161019 beta we also included more logging statements so hopefully there are some hints in the client log :)

**Comment by kvoit** [comment]
@AndyScherzinger Should I find this under Einstellungen->Protokolle (Settings->Protocols, which is directly below "Feedback", in case it is not named Protocols in English)? Because that is still completely empty for me.

**Comment by AndyScherzinger** [comment]
Yes that is exactly the one (in English it says Logs, so yes I guess it has been translated to Protokolle). It shouldn't be empty though. @tobiasKaminsky I thought the logging issue has been fixed with the latest beta (just asking, since I am unsure).

**Comment by Henry78** [comment]
The latest beta with Google playstore Beta and on https://download.nextcloud.com/android/ is dated 20160920.
How do I install the mentioned 20161019?

**Comment by AndyScherzinger** [comment]
Hi @Henry78,

the (real) "beta" is only available via via f-droid: https://f-droid.org/repository/browse/?fdfilter=com.nextcloud&fdid=com.nextcloud.android.beta and leading edge. The Google Play "beta" is rather a release candidate but unfortunately Google simply names such versions beta/alpha. The beta available via f-droid can be installed in parallel to the stable release or "beta" from Google Play.

**Comment by Henry78** [comment]
Thank you for clarifying this, @AndyScherzinger

**Comment by bennnn** [comment]
Just wanted to add I'm running the beta (20161019) and this issue is no longer present, so thank you very much for that. Funnily enough I'm using the OnePlus 3 as well, but that just seems coincidental now?

**Comment by bennnn** [comment]
Just wanted to chip in, another OnePlus 3 running beta (20161019) and no longer seeing this issue. Thank you very much for the fix!

**Comment by kvoit** [comment]
We also had issues on the Xperia Z3 of my wife, so it is not just a Oneplus thing. Perhaps it's biased, since after Google Phones got expensive and boring, the Oneplus 3 might be THE phone for people who like it open, are affine to tech stuff and thus willing to commit to a bug tracker like this. :)

Anyway, thanks again for the great job you guys are doing here. Looking forward to see client logs to resolve the last few issues.

**Comment by AndyScherzinger** [comment]
Thanks @kvoit! Unfortunately we still haven't found the issue with the client logs but @tobiasKaminsky might upgrade to a OnePlus3 so we will have a device to test/debug this matter :smiley:

**Comment by kvoit** [comment]
Is it really just a Oneplus thing? I'll check the Z3 tonight.

**Comment by AndyScherzinger** [comment]
It's not but Tobias and I only have devices without sd cards and the logs work on our phones. So we haven't been able to reproduce it yet. With a OnePlus in our hands or rather Tobias' hands we will be able to reproduce it since we know it fails on the OnePlus among others.

**Comment by propyless** [comment]
Hello,

Was also seeing the same issue reported in this issue. However, running the beta version mentioned here (20161019) works. 
However.. the first sync does not (always) work. I have to retry the failed uploads and only then does it get successfully pushed. The error I'm seeing is "Connection Error". I don't have any further logs than that.

I've got a OP3 and also tested on a Xperia z5 compact, both of which get the same issue and running Android 6.0.1.

**Comment by AndyScherzinger** [comment]
@propyless what do you mean by 
> Also tested downgrading to 1.2.0 of the nextcloud app and it works great so I'm running that again.

Do you imply the instant upload works on 1.2.0 for the phones you mentioned?

**Comment by propyless** [comment]
That's correct yes. Running version 1.2.0 of the nextcloud app the instant
upload works.

**Comment by propyless** [comment]
Have you been able to confirm if the v1.2.0 of the app works for you on the op3 or z5? @AndyScherzinger

**Comment by AndyScherzinger** [comment]
@propyless I haven't found any changes yet that would hint to this matter in the code. As for OP3 and Z5, I don't have any of these at my disposal so I can't try it out myself. :(

**Comment by propyless** [comment]
Ah I see. Well something that was implemented in 1.3.0 by reading the release notes is `Resume chunked uploads instead of complete restarts`

could this be related? as it was mentioned above something about chunk sizes not being correct.

**Comment by propyless** [comment]
@kvoit  @bennnn  if you have time over could you try the 1.2.0 version of the nextcloud app and see if that works on the op3?

**Comment by AndyScherzinger** [comment]
Hi @propyless less that is the change that introduced the error initially reported and that will be fixed in the upcoming release. From your comments I read that Instant Uploads don't seem to work, but uploading the same file works manually and also that retrying the uploads helps. The mentioned chunked uploads change imho leads to a file not being able to be uploaded at all since the chunk issue for a file remains.
Could you try out the beta and check if you see something in the log (to be found on the settings screen): https://f-droid.org/repository/browse/?fdfilter=com.nextcloud&fdid=com.nextcloud.android.beta 
The beta can be installed in parallel, so no need to uninstall the stable release.

**Comment by propyless** [comment]
Hey @AndyScherzinger ,
I have a feelign we are misunderstanding each other so heres a clarififcation just to make sure that we both understand each other :P.

----
**using the stable 1.2.0** release the auto upload sync works all the time.

**using the stable release 1.3.1** then the auto upload sync does not work at all. Not even if I retry the upload. Manual uploads work fine.

**using the beta (20161019)** the auto upload does not work consistently. but will usually work after a retry of the upload.

-----

None the less I can try the latest beta and see how that functions.

**Comment by kvoit** [comment]
Stupid question: Where do I get the 1.2.0 version? Play store gives me 1.3.1, f-droid goes back to 1.3.0RC2, but not 1.2.0.

I have been running 20161027 for a few days and just installed 20161102. I am not very warm with new upload feature, yet, though. I cannot see a way to differentiate between photos (which my volume can take and I want saved immediately) and videos (which my volume cannot take and have to wait for wifi) anymore. Apparently, an upload doesn't even trigger, but I have to investigate that further - and it is also not a thing for this issue, since uploads do not even start (queue remains empty).

**Comment by kvoit** [comment]
Stupid question: Where do I get the 1.2.0 version? Play store gives me 1.3.1, f-droid goes back to 1.3.0RC2, but not 1.2.0.

I have been running 20161027 for a few days and just installed 20161102. I am not very warm with new upload feature, yet, though. I cannot see a way to differentiate between photos (which my volume can take and I want saved immediately) and videos (which my volume cannot take and have to wait for wifi) anymore. Apparently, an upload doesn't even trigger, but I have to investigate that further - and it is also not a thing for this issue, since uploads do not even start (queue remains empty).

**Comment by kvoit** [comment]
Stupid question: Where do I get the 1.2.0 version? Play store gives me 1.3.1, f-droid goes back to 1.3.0RC2, but not 1.2.0.

I have been running 20161027 for a few days and just installed 20161102. I am not very warm with new upload feature, yet, though. I cannot see a way to differentiate between photos (which my volume can take and I want saved immediately) and videos (which my volume cannot take and have to wait for wifi) anymore. Apparently, an upload doesn't even trigger, but I have to investigate that further - and it is also not a thing for this issue, since uploads do not even start (queue remains empty).

**Comment by propyless** [comment]
Hey,

Check out the release page for this github repository . You can find it
there

**Comment by AndyScherzinger** [comment]
Hi @kvoit you can find all released installation packages here: https://download.nextcloud.com/android/

**Comment by gdamjan** [comment]
1.4.0rc2 from f-droid  still has this issue

**Comment by AndyScherzinger** [comment]
1.4.0 only activated the re-implemented instant upload for Android 7+ (now called auto upload since it can't be that instant anymore by Android design). The chunking issue and to some extend the big files upload issue have been fixed in the latest library version (https://github.com/nextcloud/android-library/releases/tag/1.0.9) which is NOT part of the client release 1.4.0 but is available in the beta client (https://f-droid.org/repository/browse/?fdfilter=Nextcloud&fdid=com.nextcloud.android.beta). Looking at the necessity of the a 1.4.0 stable release in the near future (anything below v1.4.0 will not be able to lo into an Server instance version 11+). So we might be able to ship this with 1.4.1 exept if @tobiasKaminsky is deciding to put this lib update into 1.4.0 (which has some risk since it wouldn't be tested for long in the release candidates) -> https://github.com/nextcloud/android/pull/394

**Comment by AndyScherzinger** [comment]
Further fixes via #394 have been added to 1.4.0 RC4 and should be available soon.

**Comment by joekerna** [comment]
On 1.4.0 (Huawei P9) this issue still exists

**Comment by chichin79** [comment]
Latest nextcloud app, problem Still exists on Huawei mate 9 pro

**Comment by janvonde** [comment]
and one 1+3 for me as well...

**Comment by zenlord** [comment]
I have experienced this chunking-issue as well, from the moment I started experimenting with OwnCloud 8 and still after yesterday's migration to Nextcloud 11.0.1. My smartphone is a Sony Xperia Z5 compact, and it seems that the issue is gone now that I switched to the 'Open Camera'-app on Android instead of the stock 'Sony Camera'-app. I don't want to cheer to early though, because the chunking issue was not consistently present.
As I read experiences in this thread with OnePlus and Sony, this might be a clientside issue? HTH.

**Comment by mario** [comment]
@zenlord is this with auto upload only? cc @chichin79  @janvonde @joekerna

**Comment by zenlord** [comment]
I'm using the Android client v1.4.1 from Google's Playstore. If I understand correctly what's been said in this thread, my Android client doesn't use 'Autoupload', but is still on 'InstantUpload' (whiche seems to be confirmed because I don't have the 'Autoupload'-button in my client's menu. FWIW: My Z5compact has not yet received the OTA update to Android 7.

**Comment by mario** [comment]
@zenlord you're right - gimme a few days to fix some bugs in the beta APK, and I'll let you know to install it - that one should have Auto upload ready for you even on Android 6 ;)

**Comment by mario** [comment]
Folks, you can try this out and let me know if this is better:

https://github.com/nextcloud/android/raw/beta/apks/latest.apk

Keep in mind you'll have to configure Auto upload first :)

**Comment by SuicidalTissue** [comment]
Also getting this issue with the latest APK on a oneplus3- also getting a 404, looks to happen every other upload, first goes through fine, second fails almost always. 

`51.xxx - Alex [19/Feb/2017:20:11:56 +0000] "HEAD /remote.php/webdav/neptune_share/PhoneSync/Photos/ HTTP/1.1" 200 0 "-" "Mozilla/5.0 (Android) nextcloud-android/20170214"
51.xxx - Alex [19/Feb/2017:20:11:57 +0000] "HEAD /remote.php/webdav/neptune_share/PhoneSync/Photos//IMG_20170218_145310.jpg HTTP/1.1" 404 0 "-" "Mozilla/5.0 (Android) nextcloud-android/20170214"
5`

**Comment by zenlord** [comment]
@mario @SuicidalTissue I can confirm, the behaviour of the .apk although I have not checked the logs of my Nextcloud instance.
I have also witnessed behaviour that an image through autoload would be re-uploaded multiple times (up to 35 times, until I uninstalled the .apk), while every other time the upload would fail (chunked at 1000KB). I must admit that I still had the original Nexctloud app installed on my Sony Xperia Z5 compact (Android 6 at that time, I updated to v7 over the weekend), but I had disabled the InstantUpload feature.

**Comment by mario** [comment]
@tobiasKaminsky I thought this issue was fixed, the chunking thing? Same for failure to upload the same file twice?
 
The multiple upload behaviour *should* be fixed though.

**Comment by AndyScherzinger** [comment]
@mario this is an issue in 1.4.1 which is used by @zenlord as for the chunking issue with being able to upload files several times https://github.com/nextcloud/android-library/pull/33 which has been merged to master but no new lib release has been done (yet) but should be done for 1.4.2 anyways so this will be fixed with 1.4.2

**Comment by mario** [comment]
@AndyScherzinger ah cool, I knew I heard this one :)

**Comment by propyless** [comment]
Hello,

I don't wanna be the bearer of bad news. But this chunking issue still seems to be affecting me on the One Plus 3 with 7.x~ android.

While uploading I see a lot of error logs with this message.
```
2017/06/04 16:22:52 [error] 5177#0: *244 connect() failed (111: Connection refused) while connecting to upstream, client: 83.x.x.x, server: cloud...., request: "PROPFIND /remote.php/dav/files/<user>/ HTTP/1.1", upstream: "fastcgi://127.0.0.1:9001", host: "cloud...."

2017/06/04 16:23:21 [info] 5177#0: *253 epoll_wait() reported that client prematurely closed connection, so upstream connection is closed too while sending request to upstream, client: 83.x.x.x, server: cloud....., request: "HEAD /remote.php/webdav/InstantUpload/Camera/2017/ HTTP/1.1", upstream: "fastcgi://127.0.0.1:9001", host: "cloud..,..."
```

Running 1.4.3 version of the nextcloud android app.

**Comment by Ricardosgeral** [comment]
I still have this problem. Can't upload anything in android client even in the latest R4 version. 

Phone Xiaomi Redmi Note 4X
Nextcloud server 12.0.2
Anyone having this problem? 
Could this be related with Let's encrypt credentials?

**Comment by bardo** [comment]
It has been working well for me for t least a couple months now with a Oneplus 3, Nextcloud server 12.0.2 with Letsencrypt certificate and latest stable version of the app.

**Comment by aslmx** [comment]
New to NC, everything seems to work fine, except uploading and downloading with latest Release from F-Droid (3.0.0) 

Also as other sin this thread i have noticed it seems to succeed for big images (4MB) but rather likely fails for small text files.

Webserver shows stuff like this:

`
 "HEAD /remote.php/webdav/documents/00002.vcf HTTP/1.1" 404 -
`

Also  i see failed downloads in the app for small text files, or for "medium" images (50Kb)

Altough from webserver perspective it seems fine, in the app the download fails:
`
GET /remote.php/webdav/Globetrotter_Gutschein.png HTTP/1.1" 200 52372
`

TL;DR: Download fails for any file so fofar, upload seems to work on Photos, but fails on small text files.

Shouldn't this have been fixed by now? or is this another bug. What information shall i provide to troubleshoot this? 

Thanks!

**Comment by tobiasKaminsky** [comment]
This is very strange as of now there is no real difference code-wise betwenn 3.0 and dev (which is our latest master) and we did not changed anything in regard of up/downloading.

**Comment by aslmx** [comment]
ähm... well...

So, because i thought maybe it was the fresh install from the dev version that fixed the issue... i re-installed the 3.0.0 as well.

Now it works. The "00002.vcf" test file that i had problems uploading previously now works like a charm to upload...

`
IP - user [13/Feb/2018:12:20:23 +0100] "HEAD /remote.php/webdav/00002.vcf HTTP/1.1" 404 -
IP - user [13/Feb/2018:12:20:23 +0100] "PUT /remote.php/webdav/00002.vcf HTTP/1.1" 201 -
`
It still has a 404 as it seems, but it doesn't stop the app from uploading it...

Also downloading medium sized PDFs that failed this morning work fine now.. 
The medium sized image that failed downloading before (see previous post) also works.

So the only thing that I changed in between installing the NC app previously (which was last week, but i just installed it and connected it to my NC instance, but did not do any file up/download or syncing) and noticing this issue is, that i changed the storage Disk image in the Libvirt environment that NC is installed in... No idea why this would be relating. I can't say under oath that i did not also change something else, but at least i couldn't remember if...
I'll report back if I find something out or if it happens again...

**Comment by tobiasKaminsky** [comment]
The first 404 is intended as we try to find out if the file already exists.
Glad to hear that everything is working now.
Please feel free to re-open if the problem occurs again.
